src/Microsoft.Diagnostics.Monitoring.WebApi/Exceptions/ExceptionsConfiguration.cs

```diff
@@ -0,0 +1,183 @@
+
+        private static bool CheckConfiguration(IExceptionInstance exception, List<ExceptionConfiguration> filterList, Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList)
+        {
+            var topFrame = exception.CallStack.Frames.Any() ? exception.CallStack.Frames.First() : null;
+
+            foreach (var configuration in filterList)
+            {
+                if (evaluateFilterList(configuration, topFrame))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal bool ShouldInclude(IExceptionInstance exception)
+        {
+            Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList = (configuration, topFrame) =>
+            {
+                bool include = true;
+                if (topFrame != null)
+                {
+                    CompareIncludeValues(configuration.MethodName, GetSimplifiedName(topFrame.MethodName), topFrame.MethodName, ref include);
+                    CompareIncludeValues(configuration.ModuleName, GetSimplifiedModuleName(topFrame.ModuleName), topFrame.ModuleName, ref include);
+                    CompareIncludeValues(configuration.ClassName, GetSimplifiedName(topFrame.ClassName), topFrame.ClassName, ref include);
+                }
+
+                CompareIncludeValues(configuration.ExceptionType, GetSimplifiedName(exception.TypeName), exception.TypeName, ref include);
+
+                return include;
+            };
+
+            return CheckConfiguration(exception, Include, evaluateFilterList);
+        }
+
+        internal bool ShouldExclude(IExceptionInstance exception)
+        {
+            Func<ExceptionConfiguration, CallStackFrame, bool> evaluateFilterList = (configuration, topFrame) =>
+            {
+                bool? exclude = null;
+                if (topFrame != null)
+                {
+                    CompareExcludeValues(configuration.MethodName, GetSimplifiedName(topFrame.MethodName), topFrame.MethodName, ref exclude);
+                    CompareExcludeValues(configuration.ModuleName, GetSimplifiedModuleName(topFrame.ModuleName), topFrame.ModuleName, ref exclude);
+                    CompareExcludeValues(configuration.ClassName, GetSimplifiedName(topFrame.ClassName), topFrame.ClassName, ref exclude);
+                    CompareExcludeValues(configuration.ExceptionType, GetSimplifiedName(exception.TypeName), exception.TypeName, ref exclude);
+                }
+                else
+                {
+                    CompareExcludeValues(configuration.ExceptionType, GetSimplifiedName(exception.TypeName), exception.TypeName, ref exclude);
+                }
+
+                return exclude ?? false;
+            };
+
+            return CheckConfiguration(exception, Exclude, evaluateFilterList);
+        }
+
+        private static string GetSimplifiedName(string name)
```

Comment:
Take a look at using and return `ReadOnlySpan<char>`; as it is now, it allocates 3 new strings for each call to this method and the only thing it is doing is slicing the original string.